### ISSUE TITLE
Be cautious to tf.nn.softmax.

### ISSUE THREAD

**Issue by taeguk** [DrawML/research-ml#11]
I think there is a bug in tf.nn.softmax. (just a my guess..)

tensorflow/tensorflow#4425
ibab/tensorflow-wavenet#13
AlphaGomoku/deep_learning#3

Be cautious to use tf.nn.softmax.
